lib/Sema/CSApply.cpp

```diff
@@ -668,7 +668,8 @@ namespace {
         opaqueType = LValueType::get(opaqueType);
 
       ASTContext &ctx = tc.Context;
-      auto archetypeVal = new (ctx) OpaqueValueExpr(base->getLoc(), opaqueType);
+      auto archetypeVal =
+          new (ctx) OpaqueValueExpr(base->getStartLoc(), opaqueType);
```

Comment:
I think what @hamishknight is suggesting makes more sense here, let's replace `SourceLoc` with `SourceRange` in `OpaqueValueExpr` because it is supposed to act as a container for the whole loaded value which means it has to cover its whole source range.

Reply:
@xedin Done!

Reply:
Thank you!